From 4de61d2b6081cad1aa235710a8ddb4989bd585b5 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:39:12 -0400
Subject: [ethernet] i40e: keep service tasks out of reset process

Message-id: <1409658042-22689-192-git-send-email-sassmann@redhat.com>
Patchwork-id: 90386
O-Subject: [RHEL7.1 PATCH 191/281] i40e: keep service tasks out of reset process
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Shannon Nelson <shannon.nelson@intel.com>

Make sure the service tasks don't try to meddle with the
device while a reset is in progress.  Odd things can happen
such as funky stats values.

Change-ID: I6929cb9d6d96839c9279362ca7c0e3fe6c8fcc66
Signed-off-by: Shannon Nelson <shannon.nelson@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit e57a2fea8787dffcbdff3f7f0567a1e8555dd93a)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_main.c b/drivers/net/ethernet/intel/i40e/i40e_main.c
index 25672b3..7994692 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_main.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_main.c
@@ -5961,6 +5961,12 @@ static void i40e_service_task(struct work_struct *work)
        service_task);
  unsigned long start_time = jiffies;
 
+ /* don't bother with service tasks if a reset is in progress */
+ if (test_bit(__I40E_RESET_RECOVERY_PENDING, &pf->state)) {
+  i40e_service_event_complete(pf);
+  return;
+ }
+
  i40e_reset_subtask(pf);
  i40e_handle_mdd_event(pf);
  i40e_vc_process_vflr_event(pf);
-- 
1.7.1